app/Policies/SettingsPolicy.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Policies;
+
+use App\Models\User;
+
+/**
+ * Most of those will return false as they are handle by the before()
+ * which checks for admin rights.
+ */
```

Comment:
I am not sure what you intent.

 1. Do you want to _add_ the check for being an admin inside the individual checks, too, in _addition_ to the check in `before()`?
 2. Or do you want to _move_ the check for being an admin into the individual methods and _remove_ `before()`?

I am strictly against the first one. The first approach would make `before()` redundant and duplicate code. This would be error-prone, because you always have to keep two places in sync. Just imagine that (some day in the future) we want to remove the check for being admin and someone removes the check from `before()`. It won't change anything, because the same check is performed a second time in the individual functions. That would be really confusing.

I could live with the second solution, i.e. move the check for being an admin in every single method and abandon `before()`. This way each check would be totally independent of the other checks and they can be modified individually, although it yields a more verbose code at the moment and would be inconsistent to the other policies. I would leave that decision to @ildyria.

Reply:
> I don't really care if 1 or 2, but it shouldn't just return false. It should still check if you're an admin.

I disagree with your point of view. (1) is purposely adding dead-code or (2) is adding duplicate code (which is pretty much a bad code smell).

Adding a small doc-block that says that we use `before()` to handle the admin case in front of each function is way more explicit and avoids code duplication.

Furthermore the advantage of Laravel's policies is to use the leverage of `before()` to especially take care of those overloading use-cases.